tests/main/preseed/task.yaml

```diff
@@ -19,7 +19,13 @@ prepare: |
   #shellcheck source=tests/lib/preseed.sh
   . "$TESTSLIB/preseed.sh"
 
-  wget "$(nested_get_image_url_for_vm)" -O cloudimg.img
+  # qemu-utils is not installed by default on google images which are not
+  # updated by snapd like the current 21.04 image.
+  if ! dpkg -l qemu-utils; then
```

Comment:
this needs to be done in tests/main/preseed-reset too, see: 

```
Error: 2021-02-22 22:19:55 Error preparing google:ubuntu-21.04-64:tests/main/preseed-reset (feb222102-868992) : 
-----
+ /home/gopath/src/github.com/snapcore/snapd/tests/lib/prepare-restore.sh --prepare-project-each
+ set -e
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/quiet.sh
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/dirs.sh
++ export SNAP_MOUNT_DIR=/snap
++ SNAP_MOUNT_DIR=/snap
++ export LIBEXECDIR=/usr/lib
++ LIBEXECDIR=/usr/lib
++ export MEDIA_DIR=/media
++ MEDIA_DIR=/media
++ case "$SPREAD_SYSTEM" in
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/pkgdb.sh
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/quiet.sh
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/random.sh
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/state.sh
++ SNAPD_STATE_PATH=/home/gopath/src/github.com/snapcore/snapd/tests/snapd-state
++ SNAPD_STATE_FILE=/home/gopath/src/github.com/snapcore/snapd/tests/snapd-state/snapd-state.tar
++ RUNTIME_STATE_PATH=/home/gopath/src/github.com/snapcore/snapd/tests/runtime-state
++ SNAPD_ACTIVE_UNITS=/home/gopath/src/github.com/snapcore/snapd/tests/runtime-state/snapd-active-units
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/dirs.sh
+++ export SNAP_MOUNT_DIR=/snap
+++ SNAP_MOUNT_DIR=/snap
+++ export LIBEXECDIR=/usr/lib
+++ LIBEXECDIR=/usr/lib
+++ export MEDIA_DIR=/media
+++ MEDIA_DIR=/media
+++ case "$SPREAD_SYSTEM" in
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/systemd.sh
+ case "$1" in
+ prepare_project_each
+ dmesg -c
+ fixup_dev_random
+ mv /dev/random /dev/random.orig
+ mknod /dev/random c 1 9
+ kill_gpg_agent
+ pkill -9 -e gpg-agent
+ true
+ /home/gopath/src/github.com/snapcore/snapd/tests/lib/prepare-restore.sh --prepare-suite-each
+ set -e
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/quiet.sh
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/dirs.sh
++ export SNAP_MOUNT_DIR=/snap
++ SNAP_MOUNT_DIR=/snap
++ export LIBEXECDIR=/usr/lib
++ LIBEXECDIR=/usr/lib
++ export MEDIA_DIR=/media
++ MEDIA_DIR=/media
++ case "$SPREAD_SYSTEM" in
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/pkgdb.sh
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/quiet.sh
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/random.sh
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/state.sh
++ SNAPD_STATE_PATH=/home/gopath/src/github.com/snapcore/snapd/tests/snapd-state
++ SNAPD_STATE_FILE=/home/gopath/src/github.com/snapcore/snapd/tests/snapd-state/snapd-state.tar
++ RUNTIME_STATE_PATH=/home/gopath/src/github.com/snapcore/snapd/tests/runtime-state
++ SNAPD_ACTIVE_UNITS=/home/gopath/src/github.com/snapcore/snapd/tests/runtime-state/snapd-active-units
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/dirs.sh
+++ export SNAP_MOUNT_DIR=/snap
+++ SNAP_MOUNT_DIR=/snap
+++ export LIBEXECDIR=/usr/lib
+++ LIBEXECDIR=/usr/lib
+++ export MEDIA_DIR=/media
+++ MEDIA_DIR=/media
+++ case "$SPREAD_SYSTEM" in
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/systemd.sh
+ case "$1" in
+ prepare_suite_each full
+ local variant=full
+ tests.backup prepare
tar: Removing leading `/' from member names
+ [[ ubuntu-21.04-64 == debian-sid* ]]
+ echo -n 'google:ubuntu-21.04-64:tests/main/preseed-reset '
+ [[ full = full ]]
+ /home/gopath/src/github.com/snapcore/snapd/tests/lib/reset.sh --reuse-core
active
Ensure the service is active before stopping it
Warning: Stopping snapd.service, but it can still be activated by:
  snapd.socket
active
Ensure the service is active before stopping it
+ set -e
+ [  = powerpc ]
+ [ remove = remove ]
+ grep -vF snap.mount.service
+ cut -f1 -d  
+ grep ^snap\.
+ systemctl list-unit-files --full
+ true
+ units=
+ grep -E ^snap\..*\.(service|timer|socket)$
+ echo 
+ true
+ tostop=
+ set -e
+ [  = powerpc ]
+ [ purge = purge ]
+ grep -q  /snap /snap /proc/self/mountinfo
+ grep -q  /var/snap /var/snap /proc/self/mountinfo
+ grep -vF snap.mount.service
+ cut -f1 -d  
+ grep ^snap[-.]
+ systemctl list-unit-files --full
+ units=snap-core-10893.mount
snap-test\x2dsnapd\x2dtools-x1.mount
+ grep ^snap[-.].*\.mount$
+ echo snap-core-10893.mount
snap-test\x2dsnapd\x2dtools-x1.mount
+ mounts=snap-core-10893.mount
snap-test\x2dsnapd\x2dtools-x1.mount
+ grep ^snap[-.].*\.service$
+ echo snap-core-10893.mount
snap-test\x2dsnapd\x2dtools-x1.mount
+ true
+ services=
+ grep -q What=/var/lib/snapd/snaps/ /etc/systemd/system/snap-core-10893.mount
+ echo Stopping snap-core-10893.mount
Stopping snap-core-10893.mount
+ systemctl_stop snap-core-10893.mount
+ unit=snap-core-10893.mount
+ echo Stopping unit snap-core-10893.mount
Stopping unit snap-core-10893.mount
+ systemctl stop -q snap-core-10893.mount
+ seq 20
+ echo Waiting until unit snap-core-10893.mount is stopped [attempt 1]
Waiting until unit snap-core-10893.mount is stopped [attempt 1]
+ systemctl is-active -q snap-core-10893.mount
+ echo snap-core-10893.mount is stopped.
snap-core-10893.mount is stopped.
+ return
+ cut -f3 -d/
+ grep Where=/snap/ /etc/systemd/system/snap-core-10893.mount
+ snap=core
+ cut -f4 -d/
+ grep Where=/snap/ /etc/systemd/system/snap-core-10893.mount
+ rev=10893
+ [ -n core ]
+ echo Removing snap core and revision 10893
Removing snap core and revision 10893
+ [ -d /snap/bin ]
+ find /snap/bin -maxdepth 1 -lname core -delete
+ find /snap/bin -maxdepth 1 -lname core.* -delete
+ rm -f /snap/bin/core
+ rm -f /snap/bin/core.*
+ umount -d -l /snap/core/10893
+ true
+ rm -rf /snap/core/10893
+ rm -f /snap/core/current
+ rm -rf /var/snap/core/10893
+ rm -rf /var/snap/core/common
+ rm -f /var/snap/core/current
+ [ -d /snap/core ]
+ rmdir --ignore-fail-on-non-empty /snap/core
+ [ -d /var/snap/core ]
+ find /etc/udev/rules.d -name *-snap.core.rules -execdir rm -f {} ;
+ [ -d /etc/dbus-1/system.d ]
+ find /etc/dbus-1/system.d -name snap.core.*.conf -execdir rm -f {} ;
+ rm -f /etc/modules-load.d/snap.core.conf
+ read -r f
+ find /etc/systemd/system -name snap.core.*.timer -o -name snap.core.*.socket
+ read -r f
+ find /etc/systemd/user -name snap.core.*.timer -o -name snap.core.*.socket -o -name snap.core.*.service
+ echo Removing snap-core-10893.mount
Removing snap-core-10893.mount
+ rm -f /etc/systemd/system/snap-core-10893.mount
+ rm -f /etc/systemd/system/multi-user.target.wants/snap-core-10893.mount
+ grep -q What=/var/lib/snapd/snaps/ /etc/systemd/system/snap-test\x2dsnapd\x2dtools-x1.mount
+ echo Stopping snap-test\x2dsnapd\x2dtools-x1.mount
Stopping snap-test\x2dsnapd\x2dtools-x1.mount
+ systemctl_stop snap-test\x2dsnapd\x2dtools-x1.mount
+ unit=snap-test\x2dsnapd\x2dtools-x1.mount
+ echo Stopping unit snap-test\x2dsnapd\x2dtools-x1.mount
Stopping unit snap-test\x2dsnapd\x2dtools-x1.mount
+ systemctl stop -q snap-test\x2dsnapd\x2dtools-x1.mount
+ seq 20
+ echo Waiting until unit snap-test\x2dsnapd\x2dtools-x1.mount is stopped [attempt 1]
Waiting until unit snap-test\x2dsnapd\x2dtools-x1.mount is stopped [attempt 1]
+ systemctl is-active -q snap-test\x2dsnapd\x2dtools-x1.mount
+ echo snap-test\x2dsnapd\x2dtools-x1.mount is stopped.
snap-test\x2dsnapd\x2dtools-x1.mount is stopped.
+ return
+ cut -f3 -d/
+ grep Where=/snap/ /etc/systemd/system/snap-test\x2dsnapd\x2dtools-x1.mount
+ snap=test-snapd-tools
+ cut -f4 -d/
+ grep Where=/snap/ /etc/systemd/system/snap-test\x2dsnapd\x2dtools-x1.mount
+ rev=x1
+ [ -n test-snapd-tools ]
+ echo Removing snap test-snapd-tools and revision x1
Removing snap test-snapd-tools and revision x1
+ [ -d /snap/bin ]
+ find /snap/bin -maxdepth 1 -lname test-snapd-tools -delete
+ find /snap/bin -maxdepth 1 -lname test-snapd-tools.* -delete
+ rm -f /snap/bin/test-snapd-tools
+ rm -f /snap/bin/test-snapd-tools.block /snap/bin/test-snapd-tools.cat /snap/bin/test-snapd-tools.cmd /snap/bin/test-snapd-tools.echo /snap/bin/test-snapd-tools.env /snap/bin/test-snapd-tools.fail /snap/bin/test-snapd-tools.head /snap/bin/test-snapd-tools.sh /snap/bin/test-snapd-tools.success
+ umount -d -l /snap/test-snapd-tools/x1
+ true
+ rm -rf /snap/test-snapd-tools/x1
+ rm -f /snap/test-snapd-tools/current
+ rm -rf /var/snap/test-snapd-tools/x1
+ rm -rf /var/snap/test-snapd-tools/common
+ rm -f /var/snap/test-snapd-tools/current
+ [ -d /snap/test-snapd-tools ]
+ rmdir --ignore-fail-on-non-empty /snap/test-snapd-tools
+ [ -d /var/snap/test-snapd-tools ]
+ rmdir --ignore-fail-on-non-empty /var/snap/test-snapd-tools
+ find /etc/udev/rules.d -name *-snap.test-snapd-tools.rules -execdir rm -f {} ;
+ [ -d /etc/dbus-1/system.d ]
+ find /etc/dbus-1/system.d -name snap.test-snapd-tools.*.conf -execdir rm -f {} ;
+ rm -f /etc/modules-load.d/snap.test-snapd-tools.conf
+ read -r f
+ find /etc/systemd/system -name snap.test-snapd-tools.*.timer -o -name snap.test-snapd-tools.*.socket
+ read -r f
+ find /etc/systemd/user -name snap.test-snapd-tools.*.timer -o -name snap.test-snapd-tools.*.socket -o -name snap.test-snapd-tools.*.service
+ echo Removing snap-test\x2dsnapd\x2dtools-x1.mount
Removing snap-test\x2dsnapd\x2dtools-x1.mount
+ rm -f /etc/systemd/system/snap-test\x2dsnapd\x2dtools-x1.mount
+ rm -f /etc/systemd/system/multi-user.target.wants/snap-test\x2dsnapd\x2dtools-x1.mount
+ rm -f /etc/systemd/user/snapd.session-agent.socket
+ rm -f /etc/systemd/user/snapd.session-agent.service
+ rm -f /etc/systemd/user/sockets.target.wants/snapd.session-agent.socket
+ rm -f /etc/dbus-1/session.d/snapd.session-services.conf
+ rm -f /etc/dbus-1/system.d/snapd.system-services.conf
+ rm -f /snap/README
+ echo Discarding preserved snap namespaces
Discarding preserved snap namespaces
+ [ -d /run/snapd/ns ]
+ wc -l
+ find /run/snapd/ns/ -name *.mnt
+ [ 1 -gt 0 ]
+ umount -l /run/snapd/ns/test-snapd-tools.mnt
+ rm -f /run/snapd/ns/test-snapd-tools.mnt
+ find /run/snapd/ns/ ( -name *.fstab -o -name *.user-fstab -o -name *.info ) -delete
+ umount -l /run/snapd/ns/
+ [ -e /run/systemd/container ]
+ echo Final directory cleanup
Final directory cleanup
+ rm -rf /snap/bin
+ [ -d /snap/bin ]
+ rm -rf /snap
+ [ -d /snap ]
+ rm -rf /var/snap
+ [ -d /var/snap ]
+ echo Removing extra snap-confine apparmor rules
Removing extra snap-confine apparmor rules
+ rm -f /etc/apparmor.d/snap.core.*.usr.lib.snapd.snap-confine
+ echo Removing snapd cache
Removing snapd cache
+ rm -rf /var/cache/snapd/aux /var/cache/snapd/commands.db /var/cache/snapd/names /var/cache/snapd/sections
+ echo Removing snapd state
Removing snapd state
+ rm -rf /var/lib/snapd
active
HTTP/1.0 200 OK
Content-Type: application/json
Date: Mon, 22 Feb 2021 22:19:49 GMT
Content-Length: 64

{"type":"sync","status-code":200,"status":"OK","result":["TBD"]}umount: /run/snapd/ns/*.mnt: no mount point specified.
+ systemctl reset-failed systemd-journald.service
+ systemctl restart systemd-journald.service
+ /home/gopath/src/github.com/snapcore/snapd/tests/lib/tools/journal-state start-new-log
+ [[ full = full ]]
+ echo 'Install the snaps profiler snap'
Install the snaps profiler snap
+ install_snap_profiler
+ echo 'install snaps profiler'
install snaps profiler
+ '[' 0 = 1 ']'
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/prepare.sh
++ set -eux
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/dirs.sh
+++ export SNAP_MOUNT_DIR=/snap
+++ SNAP_MOUNT_DIR=/snap
+++ export LIBEXECDIR=/usr/lib
+++ LIBEXECDIR=/usr/lib
+++ export MEDIA_DIR=/media
+++ MEDIA_DIR=/media
+++ case "$SPREAD_SYSTEM" in
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/snaps.sh
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/pkgdb.sh
+++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/quiet.sh
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/state.sh
+++ SNAPD_STATE_PATH=/home/gopath/src/github.com/snapcore/snapd/tests/snapd-state
+++ SNAPD_STATE_FILE=/home/gopath/src/github.com/snapcore/snapd/tests/snapd-state/snapd-state.tar
+++ RUNTIME_STATE_PATH=/home/gopath/src/github.com/snapcore/snapd/tests/runtime-state
+++ SNAPD_ACTIVE_UNITS=/home/gopath/src/github.com/snapcore/snapd/tests/runtime-state/snapd-active-units
+++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/dirs.sh
++++ export SNAP_MOUNT_DIR=/snap
++++ SNAP_MOUNT_DIR=/snap
++++ export LIBEXECDIR=/usr/lib
++++ LIBEXECDIR=/usr/lib
++++ export MEDIA_DIR=/media
++++ MEDIA_DIR=/media
++++ case "$SPREAD_SYSTEM" in
+++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/systemd.sh
+ os.query is-classic
+ prepare_each_classic
+ mkdir -p /etc/systemd/system/snapd.service.d
+ '[' -z '' ']'
+ rm -f /etc/systemd/system/snapd.service.d/reexec.conf
+ systemctl daemon-reload
+ systemctl restart snapd
+ '[' '!' -f /etc/systemd/system/snapd.service.d/local.conf ']'
+ /home/gopath/src/github.com/snapcore/snapd/tests/lib/tools/journal-state check-log-started
+ case "$SPREAD_SYSTEM" in
+ [[ full = full ]]
+ /home/gopath/src/github.com/snapcore/snapd/tests/lib/tools/cleanup-state pre-invariant
1287
active
active
+ tests.invariant check
tests.invariant: root-files-in-home ok
tests.invariant: crashed-snap-confine ok
tests.invariant: lxcfs-mounted ok
tests.invariant: stray-dbus-daemon ok
tests.invariant: leftover-defer-sh ok
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/nested.sh
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/systemd.sh
++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/store.sh
+++ STORE_CONFIG=/etc/systemd/system/snapd.service.d/store.conf
+++ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/systemd.sh
++ NESTED_WORK_DIR=/tmp/work-dir
++ NESTED_IMAGES_DIR=/tmp/work-dir/images
++ NESTED_RUNTIME_DIR=/tmp/work-dir/runtime
++ NESTED_ASSETS_DIR=/tmp/work-dir/assets
++ NESTED_LOGS_DIR=/tmp/work-dir/logs
++ NESTED_VM=nested-vm
++ NESTED_SSH_PORT=8022
++ NESTED_MON_PORT=8888
++ NESTED_CUSTOM_MODEL=
++ NESTED_CUSTOM_AUTO_IMPORT_ASSERTION=
++ NESTED_FAKESTORE_BLOB_DIR=/tmp/work-dir/fakestore/blobs
++ NESTED_SIGN_SNAPS_FAKESTORE=false
++ NESTED_UBUNTU_IMAGE_SNAPPY_FORCE_SAS_URL=
++ nested_get_image_url_for_vm
++ [[ google == google* ]]
++ nested_get_google_image_url_for_vm
++ case "${1:-$SPREAD_SYSTEM}" in
++ echo https://storage.googleapis.com/spread-snapd-tests/images/cloudimg/hirsute-server-cloudimg-amd64.img
+ wget --quiet https://storage.googleapis.com/spread-snapd-tests/images/cloudimg/hirsute-server-cloudimg-amd64.img -O cloudimg.img
+ mkdir -p /mnt/cloudimg
+ . /home/gopath/src/github.com/snapcore/snapd/tests/lib/preseed.sh
++ pwd
+ mount_ubuntu_image /home/gopath/src/github.com/snapcore/snapd/tests/main/preseed-reset/cloudimg.img /mnt/cloudimg
+ local CLOUD_IMAGE=/home/gopath/src/github.com/snapcore/snapd/tests/main/preseed-reset/cloudimg.img
+ local IMAGE_MOUNTPOINT=/mnt/cloudimg
+ grep nbd
+ lsmod
+ modprobe nbd
++ command -v qemu-nbd
+ systemd-run --system --service-type=forking --unit=qemu-nbd-preseed.service '' --fork -c /dev/nbd0 /home/gopath/src/github.com/snapcore/snapd/tests/main/preseed-reset/cloudimg.img
Failed to find executable : No such file or directory
-----
.
```